src/libs/SidebarUtils.ts

```diff
@@ -102,42 +107,44 @@ function setWithLimit(map, key, value) {
 let hasInitialReportActions = false;
 
 /**
- * @param {String} currentReportId
- * @param {Object} allReportsDict
- * @param {Object} betas
- * @param {String[]} policies
- * @param {String} priorityMode
- * @param {Object} allReportActions
- * @returns {String[]} An array of reportIDs sorted in the proper order
+ * @returns An array of reportIDs sorted in the proper order
  */
-function getOrderedReportIDs(currentReportId, allReportsDict, betas, policies, priorityMode, allReportActions) {
+function getOrderedReportIDs(
+    currentReportId: string | null,
+    allReports: Record<string, Report>,
+    betas: Beta[],
+    policies: Record<string, Policy>,
+    priorityMode: ValueOf<typeof CONST.PRIORITY_MODE>,
+    allReportActions: Record<string, ReportAction[]>,
+): string[] {
     // Generate a unique cache key based on the function arguments
     const cachedReportsKey = JSON.stringify(
-        // eslint-disable-next-line es/no-optional-chaining
-        [currentReportId, allReportsDict, betas, policies, priorityMode, allReportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${currentReportId}`]?.length || 1],
-        (key, value) => {
+        [currentReportId, allReports, betas, policies, priorityMode, allReportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${currentReportId}`]?.length || 1],
+        (key, value: unknown) => {
             /**
              *  Exclude 'participantAccountIDs', 'participants' and 'lastMessageText' not to overwhelm a cached key value with huge data,
              *  which we don't need to store in a cacheKey
              */
             if (key === 'participantAccountIDs' || key === 'participants' || key === 'lastMessageText') {
                 return undefined;
             }
+
             return value;
         },
     );
 
     // Check if the result is already in the cache
-    if (reportIDsCache.has(cachedReportsKey) && hasInitialReportActions) {
-        return reportIDsCache.get(cachedReportsKey);
+    const cachedIDs = reportIDsCache.get(cachedReportsKey);
```

Comment:
Are you sure about this one? Map.has() checks if the value exists, but if `cachedReportsKey` turns out to be falsey the if statement won't go through in case of Map.get(). This changes the logic of this function so I just want to make sure you know about it :)

Reply:
Good point, but map values are `Array<string>` so we should be fine :)